### ISSUE TITLE
UART Schematic: Pullup needed?

### ISSUE THREAD

**Issue by mihaigalos** [KivApple/dell-charger-emulator#5]
Hi, not sure the schematic in its current form would work:
```
TXD --------|<|----*-------- ID
                   |
RXD ----------------

GND ------------------------ GND
```
Seems `TXD` is at least missing a pullup on the anode (assuming the diode polarity is correct), since when you want to send, you would drive the pin low and open/close the diode:

```
                        VCC 
                        _|_
                       |  |
                       |__|
                        |
TXD --------|<|----*----*--- ID
                   |
RXD ----------------

GND ------------------------ GND
```

**Comment by lgolebio** [comment]
Yes. I can confirm that pullup resistor is needed 4,7Kohm. Without it it reads data but incorrect. 

![20220421_202910](https://user-images.githubusercontent.com/33582321/164530187-f24cd771-d69b-4647-8fd1-1639c24c7d5f.jpg)

**Comment by rahmajayagrs** [comment]
read correct , but write fail
![dell univ](https://user-images.githubusercontent.com/23740212/190529365-57880cc0-89df-45ac-a65b-787732ea88aa.png)

**Comment by mihaigalos** [comment]
@rahmajayagrs, so the conclusion is that it only works without the pullup?

**Comment by rahmajayagrs** [comment]
with or without pullUp still can read but can't write,
i am using ch341A

**Comment by mihaigalos** [comment]
@rahmajayagrs the value of the pullup is important, as is the maximum baud rate.
Try using a lower baudrate (i.e. `9600`) and a resistor value of `4.7K`.

Also, double check your wiring.

**Comment by arekm** [comment]
Here I was able to read EEPROM contents of original 130W Dell charger and also of emulated EEPROM but only with 4.7k pullup. No speed change was needed. My USB adapter had FTDI chip in it.

Didn't try to reprogram emulated EEPROM that way since Dell XPS 9530 I was testing recognized emulated charger properly. Programming was done in place on digispark attiny85 board, using TL866II Plus with SOIC8 clip. Fuse bits set the same way as in project Makefile.

The bad news is that emulation software doesn't work reliably. Very often dell doesn't detect the charger :/ Don't know why.
